util/lua/lua.go

```diff
@@ -0,0 +1,179 @@
+package lua
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gobuffalo/packr"
+	log "github.com/sirupsen/logrus"
+	"github.com/yuin/gopher-lua"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	luajson "layeh.com/gopher-json"
+
+	appv1 "github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+	"github.com/argoproj/argo-cd/util/settings"
+)
+
+const (
+	incorrectReturnType              = "expect table output from Lua script, not %s"
+	invalidHealthStatus              = "Lua returned an invalid health status"
+	resourceCustomizationBuiltInPath = "../../resource_customizations"
+	healthScript                     = "health.lua"
+)
+
+var (
+	box packr.Box
+)
+
+func init() {
+	box = packr.NewBox(resourceCustomizationBuiltInPath)
```

Comment:
One thing that we need to rethink, is that packing this directory includes all files (including testdata) into the binary, not just the lua scripts. Though it's just text, it will add to the size of the final binary. We should think of ways to box the lua scripts, but not the test files. This can be addressed later.